packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -4268,6 +4208,26 @@ function getWindowImpl(initialUrl: string, isHash = false): Window {
   return dom.window as unknown as Window;
 }
 
+// Fix jsdom SubmitTarget.submitter, until https://github.com/jsdom/jsdom/pull/3481 is merged
+function fixSubmitEventSubmitter(window: Window) {
+  let maybeSubmitter: HTMLInputElement | HTMLButtonElement | undefined;
+  window.addEventListener("click", (event) => {
+    if (isSubmitterElement(event.target)) maybeSubmitter = event.target;
+    setImmediate(() => {
+      // if we don't imminently trigger a submit event, then forget it
+      maybeSubmitter = undefined;
+    });
+  });
+  window.addEventListener(
+    "submit",
+    (event: any) => {
+      if (maybeSubmitter?.form === event.target)
+        event.submitter = maybeSubmitter;
```

Comment:
Is it worth making this polyfill-esque so it becomes a no-op once this is fixed upstream?

```js
      if (maybeSubmitter?.form === event.target && !event.hasOwnProperty('submitter'))
        event.submitter = maybeSubmitter;
```

Or maybe a more lax version that would still overwrite `undefined`/`null` values:

```js
      if (maybeSubmitter?.form === event.target)
        event.submitter = event.submitter || maybeSubmitter;
```

Reply:
👍  I like the polyfill idea, I'll tweak this a bit

Reply:
Done